service-worker.js

```diff
@@ -187,14 +187,20 @@ let fetchCaptureEnabled = false;
 /**
  * Intercept selected Fetch requests from the browser window
  */
- self.addEventListener('fetch', function (event) {
+self.addEventListener('fetch', function (event) {
     // Only cache GET requests
     if (event.request.method !== "GET") return;
-    // Remove any querystring before requesting from the cache
-    var rqUrl = event.request.url.replace(/\?[^?]+$/i, '');
+    var rqUrl = event.request.url;
+    var urlObject = new URL(rqUrl);
+    // Test the URL with querystring removed (hashes are not relevant in this context)
+    var searchParam = decodeURIComponent(urlObject.search);
+    var strippedUrl = event.request.url.replace(searchParam, '');
```

Comment:
Unclear logic here. It should be safer just to use `urlObject.pathname` like we do in the `fetchUrlFromZim` function below, since we're only using this to find out which Cache to use.

Also check assumption that Cache API caches encoded URIs, so we should only fully decode URI before requesting it from the ZIM, not before requesting it from the cache.

Reply:
This assumption must be correct because Cache API caches requests and responses. A proper request cannot have a bare question mark or hash anywhere in it except to indicate a querystring or an anchor target.